### ISSUE TITLE
assertion failed: self.layout().inst_ebb(inst).is_some()

### ISSUE THREAD

**Issue by bjorn3** [CraneStation/cranelift#953]
```
function u0:0(i64, i64) -> i128 system_v {
                                ebb0(v0: i64, v1: i64):
                                    v2 -> v0
                                    v6 -> v1
                                    trap user0

                                ebb30:
@0060                               v245 = iconst.i64 0
@0060                               v246 = load.i128 v245
@0064                               v248 = bxor.i128 v374, v246
@0065                               v250, v251 = isplit.i128 v370
@0065                               v252, v253 = isplit v248
@0020                               trap user0

                                ebb45:
@008d                               v369 = iconst.i64 0
@008d                               v370 = load.i128 v369
                                    v249 -> v370
@008f                               v374 = iconst.i128 0
                                    v247 -> v374
@0060                               trap user0
}
```

```
thread 'main' panicked at 'assertion failed: self.layout().inst_ebb(inst).is_some()', cranelift-codegen/src/cursor.rs:261:9
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:214
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
   7: std::panicking::begin_panic
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/libstd/panicking.rs:411
   8: cranelift_codegen::cursor::Cursor::goto_inst
             at cranelift-codegen/src/cursor.rs:261
   9: cranelift_codegen::legalizer::legalize_function
             at cranelift-codegen/src/legalizer/mod.rs:180
  10: cranelift_codegen::context::Context::legalize
             at cranelift-codegen/src/context.rs:247
  11: cranelift_codegen::context::Context::compile
             at cranelift-codegen/src/context.rs:131
  12: cranelift_codegen::context::Context::compile_and_emit
             at cranelift-codegen/src/context.rs:104
  13: clif_util::compile::handle_module
             at src/compile.rs:68
  14: clif_util::compile::run
             at src/compile.rs:26
  15: clif_util::main
             at src/clif-util.rs:257
  16: std::rt::lang_start::{{closure}}
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/libstd/rt.rs:64
  17: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:49
  18: std::panicking::try::do_call
             at src/libstd/panicking.rs:296
  19: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:82
  20: std::panicking::try
             at src/libstd/panicking.rs:275
  21: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  22: std::rt::lang_start_internal
             at src/libstd/rt.rs:48
  23: std::rt::lang_start
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/libstd/rt.rs:64
  24: main
  25: __libc_start_main
  26: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by bjorn3** [comment]
This issue was caused by a bug in my i128 branch. Fixed in e1f9de3c2b631933e79419bd5626c3f49abaebe1.

**Comment by bnjbvr** [comment]
Thanks for letting us know!
